pkg/scheduler/internal/cache/node_tree.go

```diff
@@ -184,3 +184,10 @@ func (nt *NodeTree) Next() string {
 		}
 	}
 }
+
+// NumNodes returns the number of nodes.
```

Comment:
In the spirit of what we are doing, I think it's better to call this func as `GetNoOfNodes()`

Reply:
I tried to keep it as similar as it is right now, but if that name fits better I'll change it.
@Huang-Wei what do you think?

Reply:
Either name is fine to me.